linux_os/guide/services/ldap/openldap_server/package_openldap-servers_removed/rule.yml

```diff
@@ -36,13 +40,24 @@ references:
 ocil_clause: 'it does not'
 
 ocil: |-
+    {{% if 'ubuntu' not in product %}}
```

Comment:
While we are at it, could you replace this with macro for complete ocil?

Reply:
I don't think we can? 

As far as I read it, we only have `complete_ocil_entry_package` --- we don't have a removed form of it. Don't we need a removed form (`ocil_clause: "the package is not installed"`) to do this correctly?

Reply:
I noticed that a bunch of installed and uninstalled rules use `complete_ocil_entry_package`: 

```
linux_os/guide/services/http/disabling_httpd/package_httpd_removed/rule.yml:33:{{{ complete_ocil_entry_package(package="httpd") }}}
linux_os/guide/services/http/disabling_httpd/package_httpd_removed/rule.yml:35:{{{ complete_ocil_entry_package(package="apache2") }}}
linux_os/guide/services/routing/disabling_quagga/package_quagga_removed/rule.yml:33:{{{ complete_ocil_entry_package(package="quagga") }}}
linux_os/guide/services/base/package_abrt_removed/rule.yml:31:{{{ complete_ocil_entry_package(package="abrt") }}}
linux_os/guide/services/imap/disabling_dovecot/package_dovecot_removed/rule.yml:24:{{{ complete_ocil_entry_package(package="dovecot") }}}
linux_os/guide/services/imap/disabling_dovecot/package_dovecot_removed/rule.yml:26:{{{ complete_ocil_entry_package(package="dovecot-core") }}}
linux_os/guide/services/cron_and_at/disable_anacron/rule.yml:33:{{{ complete_ocil_entry_package(package="cronie-anacron") }}}
linux_os/guide/services/smb/disabling_samba/package_samba_removed/rule.yml:18:{{{ complete_ocil_entry_package(package="samba") }}}
linux_os/guide/services/dns/disabling_dns_server/package_bind_removed/rule.yml:31:{{{ complete_ocil_entry_package(package="bind") }}}
linux_os/guide/services/ntp/package_chrony_installed/rule.yml:32:{{{ complete_ocil_entry_package(package="chrony") }}}
linux_os/guide/services/dhcp/disabling_dhcp_server/package_dhcp_removed/rule.yml:40:{{{ complete_ocil_entry_package(package="isc-dhcp-server") }}}
linux_os/guide/services/dhcp/disabling_dhcp_server/package_dhcp_removed/rule.yml:42:{{{ complete_ocil_entry_package(package="dhcp-server") }}}
linux_os/guide/services/dhcp/disabling_dhcp_server/package_dhcp_removed/rule.yml:44:{{{ complete_ocil_entry_package(package="dhcp") }}}
linux_os/guide/services/ssh/package_openssh-clients_installed/rule.yml:23:{{{ complete_ocil_entry_package(package='openssh-clients') }}}
linux_os/guide/services/mail/package_sendmail_removed/rule.yml:36:{{{ complete_ocil_entry_package(package="sendmail") }}}
linux_os/guide/services/snmp/disabling_snmp_service/package_net-snmp_removed/rule.yml:21:{{{ complete_ocil_entry_package(package="net-snmp") }}}
linux_os/guide/services/proxy/disabling_squid/package_squid_removed/rule.yml:19:{{{ complete_ocil_entry_package(package="squid") }}}
linux_os/guide/services/nfs_and_rpc/package_nfs-utils_removed/rule.yml:26:{{{ complete_ocil_entry_package(package="nfs-utils") }}}
linux_os/guide/services/ftp/disabling_vsftpd/package_vsftpd_removed/rule.yml:37:{{{ complete_ocil_entry_package(package="vsftpd") }}}
linux_os/guide/services/obsolete/telnet/package_telnet-server_removed/rule.yml:52:{{{ complete_ocil_entry_package(package="telnet-server") }}}
linux_os/guide/services/obsolete/talk/package_talk-server_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="talk-server") }}}
linux_os/guide/services/obsolete/talk/package_talk_removed/rule.yml:30:{{{ complete_ocil_entry_package(package="talk") }}}
linux_os/guide/services/obsolete/inetd_and_xinetd/package_xinetd_removed/rule.yml:36:    {{{ complete_ocil_entry_package(package="xinetd") }}}
linux_os/guide/services/obsolete/r_services/package_rsh-server_removed/rule.yml:40:{{{ complete_ocil_entry_package(package="rsh-server") }}}
linux_os/guide/services/obsolete/tftp/package_tftp-server_removed/rule.yml:39:{{{ complete_ocil_entry_package(package="tftp-server") }}}
linux_os/guide/services/obsolete/nis/package_ypserv_removed/rule.yml:42:{{{ complete_ocil_entry_package(package="ypserv") }}}
linux_os/guide/system/selinux/package_setroubleshoot-server_removed/rule.yml:28:{{{ complete_ocil_entry_package(package="setroubleshoot-server") }}}
linux_os/guide/system/selinux/package_setroubleshoot-plugins_removed/rule.yml:27:{{{ complete_ocil_entry_package(package="setroubleshoot-plugins") }}}
linux_os/guide/system/selinux/package_policycoreutils-python-utils_installed/rule.yml:22:{{{ complete_ocil_entry_package(package='policycoreutils-python-utils') }}}
linux_os/guide/system/software/system-tools/package_abrt-addon-kerneloops_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-addon-kerneloops") }}}
linux_os/guide/system/software/system-tools/package_abrt-addon-ccpp_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-addon-ccpp") }}}
linux_os/guide/system/software/system-tools/package_abrt-plugin-sosreport_removed/rule.yml:24:{{{ complete_ocil_entry_package(package="abrt-plugin-sosreport") }}}
linux_os/guide/system/software/system-tools/package_gssproxy_removed/rule.yml:24:{{{ complete_ocil_entry_package(package="gssproxy") }}}
linux_os/guide/system/software/system-tools/package_abrt-plugin-rhtsupport_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-plugin-rhtsupport") }}}
linux_os/guide/system/software/system-tools/package_pigz_removed/rule.yml:24:{{{ complete_ocil_entry_package(package="pigz") }}}
linux_os/guide/system/software/system-tools/package_abrt-cli_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-cli") }}}
linux_os/guide/system/software/system-tools/package_abrt-plugin-logger_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-plugin-logger") }}}
linux_os/guide/system/software/system-tools/package_tuned_removed/rule.yml:27:{{{ complete_ocil_entry_package(package="tuned") }}}
linux_os/guide/system/software/system-tools/package_geolite2-city_removed/rule.yml:19:{{{ complete_ocil_entry_package(package="geolite2-city") }}}
linux_os/guide/system/software/system-tools/package_krb5-workstation_removed/rule.yml:28:{{{ complete_ocil_entry_package(package="krb5-workstation") }}}
linux_os/guide/system/software/system-tools/package_geolite2-country_removed/rule.yml:19:{{{ complete_ocil_entry_package(package="geolite2-country") }}}
linux_os/guide/system/software/system-tools/package_iprutils_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="iprutils") }}}
linux_os/guide/system/software/system-tools/package_abrt-addon-python_removed/rule.yml:25:{{{ complete_ocil_entry_package(package="abrt-addon-python") }}}
linux_os/guide/system/software/integrity/crypto/package_crypto-policies_installed/rule.yml:24:{{{ complete_ocil_entry_package(package='crypto-policies') }}}
```

But wouldn't the `ocil_clause` still need to be negated? I dunno.

Reply:
Hi, thanks for fixing as much as you could. You are right that we are a bit inconsistent when using these ocils.
Currently the complete_ocil_package macro is aimed at cases when a package should be removed.
The ocil_clause actually describes the condition when the check is failing, e.g.  package is installed but it should not be.

Reply:
Cool, I think that'll be another PR :-) 

And good reminder, I forgot about the negated nature of `ocil_clause`. Thanks @vojtapolasek!